src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageItemControl.xaml.cs

```diff
@@ -27,39 +27,19 @@ public PackageItemControl()
         private void CheckBox_Toggled(object sender, RoutedEventArgs e)
         {
             var itemCheckBox = sender as CheckBox;
-            if (itemCheckBox is null || itemCheckBox.Visibility != Visibility.Visible)
-            {
-                return;
-            }
-
-            var parentContentPresenter = VisualTreeHelper.GetParent(this) as ContentPresenter;
-            if (parentContentPresenter is null)
-            {
-                return;
-            }
-
-            var parentBorder = VisualTreeHelper.GetParent(parentContentPresenter) as Border;
-            if (parentBorder is null)
-            {
-                return;
-            }
-
-            var itemContainer = VisualTreeHelper.GetParent(parentBorder) as ListBoxItem;
+            var itemContainer = itemCheckBox?.FindParent<ListBoxItem>();
```

Comment:
Think we already discussed this code before, and I explained that I manually walked the tree here since it's a fixed structure.

Reply:
The advantage of using a method to find the parent up the tree is reducing code duplication and verbosity. This does what your code did, but without having to specify what parent objects exist along the way and check each one for null since the method does that for us.

Reply:
I understand it does that, but I don't understand the benefit given that we don't need to search arbitrarily for any control here.

Reply:
The benefit is reduced code duplication, reduced verbosity, and having a reusable method to achieve what we are doing in 2 or more places. Reducing code duplication and verbosity leads to higher code quality. As mentioned in my comment below, the intent isn't to walk an unknown or arbitrary list in this particular case, but there is the added benefit that it could be used for that. Let me know if you have further questions!